src/azure-cli/azure/cli/command_modules/appconfig/_validators.py

```diff
@@ -225,10 +226,19 @@ def validate_resolve_keyvault(namespace):
 
 
 def validate_feature(namespace):
-    if namespace.feature:
-        invalid_pattern = re.compile(r'[^a-zA-Z0-9._-]')
-        invalid = re.search(invalid_pattern, namespace.feature)
-        if invalid:
-            raise CLIError("Feature name is invalid. Only alphanumeric characters, '.', '-' and '_' are allowed.")
-    else:
-        raise CLIError("Feature name cannot be empty.")
+    if namespace.feature is not None:
+        if '%' in namespace.feature:
```

Comment:
>'%' in namespace.feature: [](http://example.com/codeflow?start=11&length=25)

Why % is not allowed in feature and key? If we don't allow it, mention the restriction in argument help message.

Reply:
Backend doesn't allow % in any key names. That's the only restriction we have to enforce now. I'll add it to help message too.